api/models/author.py

```diff
@@ -48,44 +49,82 @@ def _get_cached_author(self, instance):
             # TODO: Should be an exception
             return None
 
-    def follow(self, author):
+    def _get_author(self, instance):
+        if isinstance(instance, Author):
+            return instance
+        if isinstance(instance, CachedAuthor):
+            return Author.objects.get(id=instance.id)
+        else:
+            return None
+
+    def add_friend(self, friend):
+        # Check if the potential friend has already befriended the author
+        friend = self._get_cached_author(friend)
+
+        if not isinstance(self, Author):
+            raise Exception("Not an instance of Author.")
+
+        if self.has_friend_request_from(friend):
+            print 'has request'
+            self._friend(friend)
+            # Make friendship official
+        else:
+            print 'no reuqest'
+            self._add_request_from_and_to(friend)
+
+
+    def follow(self, friend):
+        self._add_follower(friend)
+        pass
+
+    def _add_follower(self, author):
         author = self._get_cached_author(author)
         if not self.following.filter(id=author.id).exists():
             self.following.add(author)
             self.save()
 
-    def add_pending(self, friend):
+    def _add_request_from_and_to(self, friend):
+        friend = self._get_author(friend)
+        self._add_friend_request_for(friend)
+        friend._add_friend_request_from(self)
```

Comment:
Who ever is doing the request should be adding the friend to the "pending" field rather than the request field.